gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -1200,6 +1200,7 @@ private synchronized void addFailedDag(String dagId) {
      * @param dagId
      */
     private synchronized void cleanUpDag(String dagId) {
+      this.dags.get(dagId).setFlowEvent(null);
```

Comment:
Maybe just add a comment here to explain why this is necessary

Reply:
good point, added the comment